summarize/src/analysis.rs

```diff
@@ -110,7 +110,7 @@ use std::time::SystemTime;
 /// `e4` and `e3` need to be popped in the same step.
 pub fn perform_analysis(data: ProfilingData) -> Results {
     struct PerThreadState<'a> {
-        stack: Vec<Event<'a>>,
+        stack: Vec<LightweightEvent<'a>>,
```

Comment:
Since we basically always expand events to full events here, I think, it's better to keep the stack a `Vec<Event<'a>>` and just change the `for` loop below to `for current_event in data.iter().rev().map(LightweightEvent::to_event) { ... }`.